android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/adapters/NestedAdapter.java

```diff
@@ -94,10 +94,19 @@ public NestedAdapter(@NonNull Context context, int resource, @NonNull A adapter)
   }
 
   public NestedAdapter(@NonNull Context context, int resource, @NonNull A adapter, AdapterFilter<Element, A, FilterArg> filter, FilterArg filterArg) {
-    this(context, resource, adapter, filter, filterArg, filter.selectFrom(adapter, filterArg));
+    // Ensure that the list we pass through is writeable.  Errors may result otherwise.
+    this(context, resource, adapter, filter, filterArg, new ArrayList<>(filter.selectFrom(adapter, filterArg)));
   }
 
-  public NestedAdapter(@NonNull Context context, int resource, @NonNull A adapter, AdapterFilter<Element, A, FilterArg> filter, FilterArg filterArg, List<Element> filteredList) {
+  // In order for the instance to be properly constructed, the `filteredList`
+  // parameter must be mutable.  Unfortunately, there's no way to robustly
+  // check for that, and it illegal to construct a modifiable version
+  // before the super's constructor call b/c Java limitations.
+  // So... we'll keep it locked behind 'protected' for now.
+  //
+  // Could probably throw an extra parameter on for an 'internal' version
+  // and use the current signature as a 'helper' like with the constructor above?
+  protected NestedAdapter(@NonNull Context context, int resource, @NonNull A adapter, AdapterFilter<Element, A, FilterArg> filter, FilterArg filterArg, List<Element> filteredList) {
```

Comment:
Fortunately, this fellow's not actually used outside of the class yet.  So, this is sufficient for now.

Java rule:  the `super`-constructor must be called as the first line of a derived class's constructor.  It's literally impossible to in-line wrap the list and save a reference to it here, and `ArrayAdapter` doesn't provide a method to access its internal list later.